provider_shopper/lib/main.dart

```diff
@@ -30,6 +30,7 @@ class MyApp extends StatelessWidget {
         ChangeNotifierProxyProvider<CatalogModel, CartModel>(
           create: (context) => CartModel(),
           update: (context, catalog, cart) {
+            if (cart == null) throw ArgumentError.notNull('cart');
```

Comment:
I know this could have been done by doing `cart!.catalog` below, but this way it's more explicit. This really shouldn't ever happen in practice: the `update` method is called after the `create` method.